### DISCUSSION TITLE
When calling external rspamd, the HFILTER_HOSTNAME_UNKNOWN symbol always appears

### DISCUSSION THREAD

**Discussion by eosps** [rspamd/rspamd#4779] [Q&A]
I set up a test environment. Postfix+rspamd 3.7.1 is installed on one and postfix is installed on the other. I named the postfix+rspamd host A, and the independent postfix host I named B. In addition to the default proxy, I enabled a new proxy port on host A. I configured the milter of host B to the port of host A. In this way, the two mail servers share a rspamd for filtering. The problem now is that rspamd from other mail servers sent to host A works normally. But when sending to host B, the symbol HFILTER_HOSTNAME_UNKNOWN will definitely appear. Both servers are configured with the same dns. The dns configuration of the outgoing mail server should be correct and it has the correct ptr records. How should this problem be dealt with?

posftix is v3.7

**Comment by moisseev** [comment]
Check the message source. Does it have a host name in the Received header?

**Comment by eosps** [comment]
![2](https://github.com/rspamd/rspamd/assets/82441282/00a118e4-15b7-4170-a653-889f0d984382)

The Received header allows you to see the host name of the other party. Here it is
"Received: from xmbg7.mail.qq.com (unknown [101.91.43.96])"
​

**Reply by moisseev** [reply]
Then there should be a record in the Postfix log why it couldn't resolve the host name.

**Comment by eosps** [comment]
![3](https://github.com/rspamd/rspamd/assets/82441282/0d2c57db-b932-47f9-bdb3-30b40d2903a4)

This is the relevant log fragment of postfix on host B. The problem is that host A also connects from unknown but does not have the HFILTER_HOSTNAME_UNKNOWN symbol.

**Comment by eosps** [comment]
![4](https://github.com/rspamd/rspamd/assets/82441282/290c30eb-c279-4286-9590-7c083f5895f5)

Obviously ptr can be found, but I don’t know why postfix displays connect from unknown.

**Comment by m-i-k-e-y** [comment]
Just my 2 cents (I'm no expert):

The mail came from IP 101.91.43.96 according to the log snippet from postfix - did you check that IP for PTR record @eosps?

**Comment by eosps** [comment]
![5](https://github.com/rspamd/rspamd/assets/82441282/9633d067-25d1-4ed8-9cb5-b3ca415f0478)

101.91.43.96 also works @m-i-k-e-y

**Comment by moisseev** [comment]
The sender host has multiple A records. So, Postfix cannot verify forward/reverse IP address mapping.
Just try to send a test message form another mail server.

```sh
>delv xmbg7.mail.qq.com
; unsigned answer
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.51
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.54
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.89
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.96
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.105
xmbg7.mail.qq.com.      2221    IN      A       101.91.43.121
xmbg7.mail.qq.com.      2221    IN      A       101.91.44.161
xmbg7.mail.qq.com.      2221    IN      A       101.91.44.167
xmbg7.mail.qq.com.      2221    IN      A       101.91.44.196
xmbg7.mail.qq.com.      2221    IN      A       101.91.44.204
xmbg7.mail.qq.com.      2221    IN      A       101.226.139.15
xmbg7.mail.qq.com.      2221    IN      A       101.226.139.27
xmbg7.mail.qq.com.      2221    IN      A       101.226.139.34
xmbg7.mail.qq.com.      2221    IN      A       101.226.139.58
xmbg7.mail.qq.com.      2221    IN      A       101.226.139.69
```

**Comment by eosps** [comment]
Host A acts as a relay server, and the filtered mail is then sent to server B. The postfix log in host A also contains connections from unknown. Why does rspamd in host A not show the HFILTER_HOSTNAME_UNKNOWN symbol? Their operating system environments are the same. The only difference is that host A calls local rspamd, and host B remotely calls rspamd of host A.

**Comment by moisseev** [comment]
Relaying mail is another story. In this case SMTP session information is lost. You probably should add host A to [local_addrs](https://rspamd.com/doc/configuration/options.html) on host B or disable some of the listed check selectively. Though I don't see the point in filtering the same mail twice.

**Comment by moisseev** [comment]
> Why does rspamd in host A not show the HFILTER_HOSTNAME_UNKNOWN symbol?

It falls back to own resolver.

**Reply by eosps** [reply]
Sorry, I didn't understand what you said.
Don't think about the relay service I mentioned earlier. What I want to say is why the HFILTER_HOSTNAME_UNKNOWN symbol does not appear when rspamd and postfix are on the same server. The HFILTER_HOSTNAME_UNKNOWN symbol will appear on different servers for rspamd and postfix. Of course I have to point out that most of the mail servers I communicate with have domain names that do not have ptr.

**Reply by moisseev** [reply]
Perhaps I was wrong about fallback.

So what is the problem? The HFILTER_HOSTNAME_UNKNOWN is never appears for local Postfix, even sender host does not have PTR, or it appears for remote Postfix for sender hosts that have PTRs and passed FCrDNS verification?

**Comment by eosps** [comment]
Thanks for everyone's replies, I finally turned off this symbol. I can't figure out why this problem occurs and I'm tired.
